cl/corpus_importer/management/commands/columbia_merge.py

```diff
@@ -0,0 +1,865 @@
+                    )
+                    if s.ratio() >= 0.6:
+                        # columbia names are better
+                        op.author_str = author_str
+                    else:
+                        raise AuthorException(f"Authors don't match")
+                elif any(s.isupper() for s in op.author_str.split(",")):
+                    # Some names are uppercase, update with processed names
+                    op.author_str = author_str
+
+        converted_text = convert_columbia_html(
+            file_opinion["opinion"], columbia_pos
+        )
+        op.html_columbia = str(converted_text)
+        op.save()
+
+
+def match_text_lists(
+    file_opinions_list: list[Any], cl_opinions_list: list[Any]
+) -> dict[int, int]:
+    """Generate matching lists above threshold
+    :param file_opinions_list: Opinions from file
+    :param cl_opinions_list: CL opinions
+    :return: Matches if found or empty dict
+    """
+    # We import this here to avoid a circular import
+    from cl.corpus_importer.management.commands.harvard_opinions import (
+        compare_documents,
+    )
+
+    scores = similarity_scores(file_opinions_list, cl_opinions_list)
+
+    matches = {}
+    for i, row in enumerate(scores):
+        j = row.argmax()  # type: ignore
+        # Lower threshold for small opinions.
+        # Remove non-alphanumeric and non-whitespace characters from lowercased text,
+        # this tries to make both texts in equal conditions to prove if both are
+        # similar or equal
+        file_opinion = re.sub(
+            r"[^a-zA-Z0-9 ]", "", file_opinions_list[i].lower()
+        )
+        cl_opinion = re.sub(r"[^a-zA-Z0-9 ]", "", cl_opinions_list[j].lower())
+
+        # NOTE: get_cosine_similarity works great when both texts are almost the same
+        # with very small variations
+        cosine_sim = get_cosine_similarity(file_opinion, cl_opinion)
+        # NOTE: compare_documents works good when the opinion from the file is a
+        # subset of the opinion in CL, the percentage represents how much of the
+        # opinion of the file is in the opinion from cl (content in cl opinion can
+        # have other data in the body like posture, attorneys, etc. e.g. in cluster
+        # id: 7643871 we have the posture and the opinion text but in the xml file we
+        # only have the opinion text, cosine_sim: 0.1639075094124459 and
+        # percent_match: 73)
+        percent_match = compare_documents(file_opinion, cl_opinion)
+
+        # Sometimes one algorithm performs better than the other, this is due to some
+        # additional text, such as editor's notes, or the author, page number or posture
+        # added to the opinion
+        if cosine_sim < 0.60 and percent_match < 60:
+            continue
+
+        matches[i] = j
+
+    # Key is opinion position from file, Value is opinion position from cl opinion
+    # e.g. matches {0: 1, 1: 2} 0 is file opinion and 1 in cl opinion, 1 is file
```

Comment:
all of this commenting feels out of place  - maybe this should go in the doc string explaining things.